src/container/oid_iv.c

```diff
@@ -133,12 +134,11 @@ oid_iv_ent_update(struct ds_iv_entry *ns_entry, struct ds_iv_key *iv_key,
 	entry = ns_entry->iv_value.sg_iovs[0].iov_buf;
 	rc = ABT_mutex_trylock(entry->lock);
 	/** For retry requests, from _iv_op(), the lock may not be released in some cases. */
-	if (rc == ABT_ERR_MUTEX_LOCKED && entry->current_req != src)
+	if (rc == ABT_ERR_MUTEX_LOCKED && entry->current_req != src->sg_iovs[0].iov_buf)
```

Comment:
can we use  @priv as identifier of owner? 
btw, I still don't understand why we will enter here for more than once.

Reply:
cannot use priv as it will change for every retry. there is a new PR in master https://github.com/daos-stack/daos/pull/15727 

It can enter here every time when update this IV. the update() and refresh() take/release the lock, but some err cases the update() called but  refresh() did not called so the retried case cannot take the lock again. So wangdi add https://github.com/daos-stack/daos/pull/13632 long time ago for that case.

Reply:
correct. i set as priv, but in that case, on_get is called on every retry (which IMO is not good and we should not do that), but for now it won't detect that.